### PR TITLE
docs(k8s): secure cluster with pn

### PR DESCRIPTION
**Pull request by bene2k1**
### Your checklist for this pull request

- [ ] Check that the commit messages match our requested structure.
- [ ] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description
Please describe your pull request.

### CODE REVIEW

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
+</Message>
+
+Scaleway Kubernetes Kapsule provides a managed environment to create, configure and run a cluster of preconfigured machines for containerized applications. This allows you to create Kubernetes clusters without the complexity of managing the infrastructure.
+
+All new Kubernetes clusters are deployed with a Scaleway regional Private Network. **We recommend migrating existing clusters to a regional Private Network as well, using the Scaleway console.**
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:25]:
```suggestion
All new Kubernetes clusters are deployed with a Scaleway [regional Private Network](https://www.scaleway.com/en/docs/network/vpc/concepts/#regional-private-networks). **We recommend migrating existing clusters to a regional Private Network as well, using the Scaleway console.**
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+Currently, worker nodes are still shipped with public IP addresses.
+These IPs are used for outgoing traffic from your nodes towards the Internet only; no services are listening on them by default.
+Despite having public IP addresses for limited maintenance and operational reasons, you can be assured that your cluster remains fully secure. Find more details below.
+
+<Lightbox src="kubernetes-kapsule-on-pn-v2.webp" alt="Schema Kubernetes Kapsule on Private Networks" size="large" />
+
+## Why have a Private Network for your Kubernetes Kapsule cluster?
+
+A Private Network brings additional features to your cluster, such as:
+* Implementation of best practices in terms of security: All Scaleway resources to communicate securely (Instances, Load Balancers, Managed Databases), with less surface of attacks
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:36]:
```suggestion
* Implementation of best practices in terms of security: all Scaleway resources can communicate securely (Instances, Load Balancers, Managed Databases), with less surface area for attack
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+* Compliance with expectations from the market (enterprise customers)
+* Less manual configuration work such as security group configuration, IP range configuration etc.
+* Lower latency
+
+You can create new regional Private Networks via the Scaleway Console and quickly add clusters to existing regional Private Networks using the Scaleway Console. Each cluster is autoconfigured using a /22 IP subnet thanks to Scaleway’s IPAM.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:41]:
```suggestion
You can use the Scaleway console to both create new regional Private Networks, and add clusters to existing regional Private Networks. Each cluster is autoconfigured using a /22 IP subnet thanks to Scaleway’s IPAM.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+## Why do worker nodes still have public IPs?
+
+Public IP addresses are currently required to maintain the administration of your worker nodes and their control plane to ensure the full functionality of your clusters. 
+Do note, however, that incoming traffic on these public IPs is blocked by default thanks to a security group set at DROP ALL, and only outgoing traffic necessary for Kubernetes to run (e.g., apt, Docker image pull, etc.) is allowed. 
+
+## Can I use a Public Gateway with my Private Network to exit all outgoing traffic from the nodes?
+
+At this moment, outgoing traffic is through the public IP of each node.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:50]:
```suggestion
At the moment, outgoing traffic is through the public IP of each node.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Public Gateway will soon be compatible with IPAM and thus supported by Kapsule in a future release.
+
+## How can I migrate my existing clusters to regional Private Networks?
+
+You can use the Scaleway console to migrate all your previous clusters (recommended).
+
+1. In Kubernetes, locate your public clusters in the list (Network column)
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:57]: I think these steps could be clearer, not sure what is meant by "Network column"?

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+2. Click the clusters name to navigate to your cluster information
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:58]:
```suggestion
2. Click the cluster's name to navigate to your cluster information
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+3. Use the recommended migration action in your cluster information, or go to the Private Network tab to start the migration.
+    <Message type="note">
+    Migrating an existing cluster is only available for clusters with a Kubernetes version above `v1.22.0`. Also, once a cluster is migrated, you will not be able to rollback this change.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:61]:
```suggestion
    Migrating an existing cluster is only available for clusters with a Kubernetes version above `v1.22.0`. Not that once a cluster is migrated, you will not be able to roll back this change.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    </Message>
+
+What will happen after starting the migration:
+
+1. Your control plane will restart for a first time: the Kubernetes API of your cluster will temporarily be unavailable.
+2. Your pools will be upgraded to migrate the nodes into the Private Network. All of your nodes will be rebooted according to the specified Upgrade Policy of your pools.
+3. Once all your nodes have rebooted, your control-plane will be configured to use the Private Network and restarted: your Load Balancers will be reconfigured and migrated to the Private Network.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:68]:
```suggestion
3. Once all your nodes have rebooted, your control plane will be configured to use the Private Network and restarted: your Load Balancers will be reconfigured and migrated to the Private Network.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+4. The CNI of your cluster will be reconfigured and restarted to use the Private Network.
+    <Message type="important">
+      During step IV, the Pod network of your cluster will be temporarily unavailable for 1 to 10 minutes as all Pods of the CNI are restarted, depending on the size of your cluster and the CNI you are using.
+      Your Pods will be unable to communicate with each other during this step.
+    </Message>
+
+To further improve the security of your Kubernetes cluster, you can configure a Security Group to block inbound traffic on the public network interface of your nodes.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:75]: maybe link to how to do this?

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+We are currently at the first phase of an isolated Kubernetes infrastructure. In the future, additional features are planned, such as
+- compatibility with Scaleway Public Gateways, to be used as a single exit point for the cluster
+- extended VPC integration of Kubernetes Kapsule, with support of Multi-AZ features.
+- Isolation of the control plane on top of your worker nodes
+
+## Will the control plane also be located inside the Private Network?
+Currently, only worker nodes are located in the Private Network of your cluster. The communication between the nodes and the control plane uses the Public IP of the node.
+
+## Which IP ranges are used for the Private Network of my cluster?
+We automatically assign a /22 IP subnet from a private network to your cluster.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:94]:
```suggestion
We automatically assign a /22 IP subnet from a Private Network to your cluster.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+			"private_network_id": "$PRIVATE_NETWORK_ID"
+		}
+	}
+}'
+```
+Please refer to the [Managed Database for PostgreSQL and MySQL API documentation](https://www.scaleway.com/en/developers/api/managed-database-postgre-mysql/#path-endpoints-create-a-new-database-instance-endpoint) for further information.
+
+<Message type="note">
+  * This action replaces your current endpoint, which means you might need to update any environment configurations that point to the old endpoint.
+  * This feature is not yet provided in Scaleway Console.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:119]:
```suggestion
  * This feature is not yet provided in Scaleway console.
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+</Message>
+
+## Are Managed Load Balancers compatible with Kubernetes Kapsule Private Networks?
+
+Managed Load Balancers support Private Networks with private backends and public Frontends, meaning the traffic is forwarded to your worker nodes through your clusters' Private Network.
```
Comment (RoRoJ) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:124]:
```suggestion
Managed Load Balancers support Private Networks with private backends and public frontends, meaning the traffic is forwarded to your worker nodes through your clusters' Private Network.
```

**Review by RoRoJ — APPROVED** [review]

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+* Compliance with expectations from the market (enterprise customers)
+* Less manual configuration work such as security group configuration, IP range configuration etc.
+* Lower latency
+
+You can use the Scaleway console to both create new regional Private Networks, and add clusters to existing regional Private Networks. Each cluster is autoconfigured using a /22 IP subnet thanks to Scaleway’s IPAM.
+
+## Why do worker nodes still have public IPs?
+
+Public IP addresses are currently required to maintain the administration of your worker nodes and their control plane to ensure the full functionality of your clusters. 
+Do note, however, that incoming traffic on these public IPs is blocked by default thanks to a security group set at DROP ALL, and only outgoing traffic necessary for Kubernetes to run (e.g., apt, Docker image pull, etc.) is allowed.
```
Comment (nedaehm) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:46]:
```suggestion
However, incoming traffic on these public IPs is blocked by default thanks to a security group set at DROP ALL, and only outgoing traffic necessary for Kubernetes to run (e.g., apt, Docker image pull, etc.) is allowed. 
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+However, be careful when updating the Security Group: if you currently use Kubernetes services with NodePorts, need SSH access, or need ICMP access, you must add a rule to allow this traffic.
+
+## Is Kosmos compatible with Private Networks?
+
+Only Kapsule can use a Private Network.
+Kosmos uses Kilo as a CNI, which uses Wireguard to create a VPN Mesh between nodes for communication between pods. Any node in Kosmos, either in Scaleway or outside, uses these VPN tunnels to communicate securely by construct.
+
+## What future options will there be for isolation?
+
+We are currently at the first phase of an isolated Kubernetes infrastructure. In the future, additional features are planned, such as
```
Comment (nedaehm) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:85]:
```suggestion
We are currently at the first phase of an isolated Kubernetes infrastructure. In the future, additional features are planned, such as:
```

File: containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+			"private_network_id": "$PRIVATE_NETWORK_ID"
+		}
+	}
+}'
+```
+Please refer to the [Managed Database for PostgreSQL and MySQL API documentation](https://www.scaleway.com/en/developers/api/managed-database-postgre-mysql/#path-endpoints-create-a-new-database-instance-endpoint) for further information.
+
+<Message type="note">
+  * This action replaces your current endpoint, which means you might need to update any environment configurations that point to the old endpoint.
+  * This feature is not yet provided in Scaleway console.
```
Comment (nedaehm) [inline containers/kubernetes/reference-content/secure-cluster-with-private-network.mdx:119]:
```suggestion
  * This feature is not yet provided in the Scaleway console.
```

**Review by nedaehm — APPROVED** [review]

**Pull request merged**